source/common/protobuf/utility.h

```diff
@@ -177,6 +178,51 @@ class MessageUtil {
   static void loadFromYaml(const std::string& yaml, Protobuf::Message& message);
   static void loadFromFile(const std::string& path, Protobuf::Message& message);
 
+  static void checkForDeprecation(const Protobuf::Message& message, bool fatal) {
```

Comment:
Nit: prefer `warn_only` vs. `fatal`, since it's not fatal, it's raising an exception which could be handled in other ways at a call-site (even though you probably only plan on using it for fatal..).